corehq/apps/api/resources/auth.py

```diff
@@ -149,6 +150,20 @@ def dummy(request, domain, **kwargs):
             return response
 
 
+class NoAPIPermissionsAuthentication(LoginAndDomainAuthentication):
```

Comment:
nit: `OpenAPIAuthentication`?

Reply:
I am not sure if it would be valid in our case but [`OpenAPI`](https://www.openapis.org/) is generally used for standards/specs that are used to define the APIs, and `OpenAPIAuthentication` sounds like the Authentication for all the APIs?

Reply:
cool, thanks @AmitPhulera